src/routing/core/api.rs

```diff
@@ -304,23 +301,10 @@ impl Core {
         previous_name: Option<XorName>,
         destination_key: Option<bls::PublicKey>,
     ) -> Result<Vec<Command>> {
-        if peer.is_reachable() {
-            self.propose(Proposal::Online {
-                member_info: MemberInfo::joined(peer),
-                previous_name,
-                destination_key,
-            })
-        } else {
-            let variant = Variant::JoinResponse(Box::new(JoinResponse::Rejected(
-                JoinRejectionReason::NodeNotReachable(*peer.addr()),
-            )));
-
-            trace!("Sending {:?} to {}", variant, peer);
-            Ok(vec![self.send_direct_message(
-                (*peer.name(), *peer.addr()),
-                variant,
-                *self.section.chain().last_key(),
-            )?])
-        }
+        self.propose(Proposal::Online {
+            member_info: MemberInfo::joined(peer),
```

Comment:
the reachable flag of this input peer was set separately at https://github.com/maidsafe/sn_routing/blob/master/src/routing/dispatcher.rs#L185

Though the chance is tiny, still better to check the flag is true here?

Meanwhile, with your current change, I think we can now have `Command::ProposeOnline` contains the peer have reachable flag set to true directly.
So that we can avoid having extra check here.

i.e. at the place at  https://github.com/maidsafe/sn_routing/blob/master/src/routing/core/messaging/handling/mod.rs#L676
the reachable flag shall be set to true already.